### DISCUSSION TITLE
PDM hatching unwanted 'eggs'

### DISCUSSION THREAD

**Discussion by justasojourner** [pdm-project/pdm#618] [Q&A]
Hi - I am not necessarily saying this is a bug — but it is unexpected behaviour. I note that this issue ticket 
https://github.com/pdm-project/pdm/issues/253 (which is closed - so solved?) mentions the same thing.

### The problem

I seem to be having Python egg folders created — [project_name].egg-info

I tracked it down to if the project name field in `pyproject.toml` is filled in:

```
[project]
name = "Delme"
version = "0.0.0"
description = "Delete me"
```

I noticed this after I added a package after I had named the project in the `pyproject.toml` file - which I thought would be a good thing to do.

I tested it - made a new project (and explicitly said 'N' to is this a library)

```
delme % pdm init
Creating a pyproject.toml for PDM...
Please enter the Python interpreter to use
0. /System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python (2.7)
1. /usr/local/bin/python3 (3.9)
2. /usr/local/bin/python3.9 (3.9)
3. /usr/local/bin/python3.8 (3.8)
4. /Library/Developer/CommandLineTools/usr/bin/python3 (3.8)
5. /usr/local/bin/venv/bin/python (3.9)
Please select: [0]: 1
Using Python interpreter: /usr/local/bin/python3 (3.9)
Is the project a library that will be uploaded to PyPI? [y/N]: N
License(SPDX name) [MIT]:
Author name []: Fred Nurk
Author email []: noone@nowwhere.com
Python requires('*' to allow any) [>=3.9]:
Changes are written to pyproject.toml.
```

Then _filled in the project details_,  **then when the very next package gets installed** I see this:

delme % pdm add timeflake
Adding packages to default dependencies: timeflake
✔ 🔒 Lock successful
Changes are written to pdm.lock.
Changes are written to pyproject.toml.
Synchronizing working set with lock file: 1 to add, 0 to update, 0 to remove

  ✔ Install timeflake 0.4.0 successful
**Installing the project as an editable package...
  ✔ Install Delme 0.0.0 successful**

🎉 All complete!

I don't want/need the Python setuptools created egg-info directory:

1. Is this as-designed behaviour?
2. How can I stop it happening, and can I still have the project named in `pyproject.toml`?
3. Can I safely delete the directory because now I see this:

```
drp % pdm list --graph
drp 0.0.0 [ required: This project ]
├── beautifulsoup4 4.9.3 [ required: ~=4.9 ]
│   └── soupsieve 2.2.1 [ required: >1.2 ]
├── openpyxl 3.0.7 [ required: ~=3.0 ]
│   └── et-xmlfile 1.1.0 [ required: Any ]
├── pandas 1.3.2 [ required: ~=1.3 ]
│   ├── numpy 1.21.1 [ required: >=1.17.3 ]
│   ├── python-dateutil 2.8.2 [ required: >=2.7.3 ]
│   │   └── six 1.16.0 [ required: >=1.5 ]
│   └── pytz 2021.1 [ required: >=2017.3 ]
├── psycopg2-binary 2.9.1 [ required: ~=2.9 ]
├── python-dotenv 0.19.0 [ required: ~=0.19 ]
├── timeflake 0.4.0 [ required: ~=0.4 ]

```

**Comment by justasojourner** [comment]
Hi all - just wanting to check, can I safely remove the unwanted package made from the project and to stop it happening again I just remove the name of the project in the `pyproject.toml` file?

**Reply by frostming** [reply]
1. Just leave `name` and `version` as empty in your project.toml.
2. No, when you don't want to publish it as a library, the name is not needed? is it?
3. You can delete the egg-info directory safely
